web-api/src/storage/elastic.rs

```diff
@@ -407,6 +408,27 @@ impl KnnSearchParams<'_> {
                 json!({ "range": { "properties.publication_date": { "gte": published_after } } }),
             );
         }
+        if let Some(opt_filter) = self.filter {
+            let opt_filter = match opt_filter {
+                Filter::Compare(compare) => match compare.operation {
+                    CompareOp::Eq => {
+                        json!({ "term": { format!("properties.{}", compare.field): compare.value } })
```

Comment:
This probably works but elasticsearch documents it as:

```json
{ "term": { "property.name": { "value": "comparision_value" } } }
```

https://www.elastic.co/guide/en/elasticsearch/reference/8.8/query-dsl-term-query.html

funny thing is with `terms` it does not do so :face_exhaling: 

```suggestion
                        json!({ "term": { format!("properties.{}", compare.field): { "value":  compare.value } } })
```